### DISCUSSION TITLE
Vitepress build vant error

### DISCUSSION THREAD

**Discussion by lyanerzhang** [vuejs/vitepress#2834] [Q&A] [answered]
### Describe the bug

node:v16.13.2
vitepress: v1.0.0-rc.4
vant: ~3.4.9
vite ^4.4.9
执行`vitepress build docs`报以下错误
![image](https://github.com/vuejs/vitepress/assets/38914404/aa7f6868-1bea-485f-b157-0b79924ac53f)

### Reproduction

// .vitepress config.ts
import { defineConfig } from 'vitepress'
import { mdPlugin } from './config/plugins'
import sidebars from "./sidebar"

export default defineConfig({
  title: "chuxin-ui-mobile",
  // description: "A VitePress Site",
  ignoreDeadLinks: true,
  base: "/cx-mobile-ui-docs/",
  head: [
    [
      "link",
      { rel: "icon", href: "/images/logo.png" }
    ]
  ],
  themeConfig: {
    logo: "/images/logo.png",
    nav: [
      { text: '首页', link: '/' },
      { text: '组件', link: '/components/install', activeMatch: '/components/' },
      { text: '指南', link: '/development/directoryIntroduction' }
    ],
    sidebar: {
      "/": sidebars
    },
    socialLinks: [
      { icon: 'github', link: 'https://github.com/vuejs/vitepress' }
    ]
  },
  markdown: {
    headers: {
      level: [0, 0],
    },
    theme: { light: 'github-light', dark: 'github-dark' },
    config: (md) => mdPlugin(md),
  }
})

### Expected behavior

`vitepress build docs` 可以build成功

### System Info

```sh
无
```

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by brc-dd** [comment, marked as answer]
Can you share it as stackblitz url? (https://vitepress.new). You most probably need to add something like this in your config file:

```ts
  vite: {
    ssr: {
      noExternal: ['vant'],
    }
  }
```

**Comment by lyanerzhang** [comment]
https://stackblitz.com/edit/vite-gd6obf?file=package.json
![image](https://github.com/vuejs/vitepress/assets/38914404/68be11c0-be16-4009-a12d-c37567979cff)

**Comment by brc-dd** [comment]
Your example doesn't appear to be complete. Can you update it to show how you're using that component.

**Comment by lixiaofa** [comment]
I also encountered the same problem , It's also related to vant
error infor: 
![image](https://github.com/vuejs/vitepress/assets/23230215/b53d3821-a178-47a0-835f-bce1ce531003)
@brc-dd

**Reply by brc-dd** [reply]
Have you tried the `noExternal` workaround I mentioned above? You might need `['vant', 'vant/es/popup/style']` there.

**Reply by lyanerzhang** [reply]
谢谢~ 已经解决上述问题了

**Reply by lixiaofa** [reply]
Still getting the same error
![image](https://github.com/vuejs/vitepress/assets/23230215/75fdbc78-6db4-47c5-87ae-8e9560d380a6)

**Reply by brc-dd** [reply]
Can you provide the repo where it's happening? Also, have you checked if vant is installed? Your error is different than what OP had.

**Reply by lixiaofa** [reply]
https://github.com/lixiaofa/fast-plus
pnpm docs:build

**Reply by lixiaofa** [reply]
installed ， Can you help me take a look? Thank you very much

**Reply by brc-dd** [reply]
Is there some minimal version of the repo? Currently it has too many things, and requires different secrets like crowdin credentials, etc. Running `docs:build` just gives me `Could not resolve "../i18n/pages/...`.

**Reply by lixiaofa** [reply]
Solved it, thank you. What I still don't understand is why as long as it's CSS, it can't solve it

**Comment by lyanerzhang** [comment]
上述问题已经解决。

**Reply by lixiaofa** [reply]
你怎么解决的 ， 大佬
